examples/nlp/multimodal_entailment.py

```diff
@@ -0,0 +1,577 @@
+    # are model-dependent, so this gets loaded from the SavedModel.
+    packer = hub.KerasLayer(
+        bert_preprocess.bert_pack_inputs,
+        arguments=dict(seq_length=seq_length),
+        name="packer",
+    )
+    model_inputs = packer(truncated_segments)
+    return keras.Model(input_segments, model_inputs)
+
+
+bert_preprocess_model = make_bert_preprocess_model(["text_1", "text_2"])
+keras.utils.plot_model(bert_preprocess_model, show_shapes=True, show_dtype=True)
+
+"""
+### Run the preprocessor on a sample input
+"""
+
+idx = np.random.choice(len(train_df))
+row = train_df.iloc[idx]
+sample_text_1, sample_text_2 = row["text_1"], row["text_2"]
+print(f"Text 1: {sample_text_1}")
+print(f"Text 2: {sample_text_2}")
+
+test_text = [np.array([sample_text_1]), np.array([sample_text_2])]
+text_preprocessed = bert_preprocess_model(test_text)
+
+print("Keys           : ", list(text_preprocessed.keys()))
+print("Shape Word Ids : ", text_preprocessed["input_word_ids"].shape)
+print("Word Ids       : ", text_preprocessed["input_word_ids"][0, :16])
+print("Shape Mask     : ", text_preprocessed["input_mask"].shape)
+print("Input Mask     : ", text_preprocessed["input_mask"][0, :16])
+print("Shape Type Ids : ", text_preprocessed["input_type_ids"].shape)
+print("Type Ids       : ", text_preprocessed["input_type_ids"][0, :16])
+
+
+"""
+We will now create `tf.data.Dataset` objects from the dataframes for performance.
+
+Note that the text inputs will be preprocessed as a part of the data input pipeline. But
+the preprocessing modules can also be a part of their corresponding BERT models. This
+helps reduce the training/serving skew and lets our models operate with raw text inputs.
+Follow [this tutorial](https://www.tensorflow.org/text/tutorials/classify_text_with_bert)
+to know more about how to incorporate the preprocessing modules directly inside the
+models. 
+"""
+
+
+def dataframe_to_dataset(dataframe):
+    columns = ["image_1_path", "image_2_path", "text_1", "text_2", "label_idx"]
+    dataframe = dataframe[columns].copy()
+    labels = dataframe.pop("label_idx")
+    ds = tf.data.Dataset.from_tensor_slices((dict(dataframe), labels))
+    ds = ds.shuffle(buffer_size=len(dataframe))
+    return ds
+
+
+"""
+### Preprocessing utilities
+"""
+
+resize = (128, 128)
+bert_input_features = ["input_word_ids", "input_type_ids", "input_mask"]
+
+
+def read_resize(image_path):
+    extension = tf.strings.split(image_path)[-1]
+
+    image = tf.io.read_file(image_path)
+    if extension == b"jpg":
+        image = tf.image.decode_jpeg(image, 3)
+    else:
+        image = tf.image.decode_png(image, 3)
+    image = tf.image.resize(image, resize)
+    return image
+
+
+def preprocess_text(text_1, text_2):
+    text_1 = tf.convert_to_tensor([text_1])
+    text_2 = tf.convert_to_tensor([text_2])
+    output = bert_preprocess_model([text_1, text_2])
+    output = {feature: tf.squeeze(output[feature]) for feature in bert_input_features}
```

Comment:
Why the squeeze?

Reply:
It is because the preprocessor adds a batch dimension which wouldn't have been a problem if we had applied the preprocessor on batches. For a single sample, the expected dimension is `(128, )` with the squeeze included. 

Since we are processing the images and the texts simultaneously I tried to keep their preprocessing steps as coherent as possible. To this end, I got confused to apply the following logic to batches:

```python
def read_resize(image_path):
    extension = tf.strings.split(image_path)[-1]

    image = tf.io.read_file(image_path)
    if extension == b"jpg":
        image = tf.image.decode_jpeg(image, 3)
    else:
        image = tf.image.decode_png(image, 3)
    image = tf.image.resize(image, resize)
    return image
```

Reply:
Do you still get this with TF nightly?